### DISCUSSION TITLE
Opportunity to know if all options are filtered out

### DISCUSSION THREAD

**Discussion by antollika** [JedWatson/react-select#5486] [Ideas]
I have a custom Menu component for my select and I have to render a custom component when all options are filtered out
`noOptionsMessage` does not suit me. I need a separate variable like `isNoOptions` to something similar.
I've also tried to check options length but it always the same even if all of them are filtered out by the input value

I have only one solution for now - filter options with `inputValue` by myself but why I should do it if select already did it?

**Comment by peyssoa** [comment]
@antollika were you able to find a solution or workaround for your problem? I've also ended up in the same place

**Comment by manjushsh** [comment]
Hi @antollika  and @peyssoa, I am not sure if I am understanding problem correctly but you want to render a custom component in place of 'No Options found' component?
If that is the case, you can use components like this right? https://codesandbox.io/s/react-select-examples-6picqk

**Reply by peyssoa** [reply]
Hi @manjushsh 
essentially, what i was looking for was a way to trigger something if the react select component has `no options` available based on the users input. That could be returning the length of the number of available items on each query (where if its 0 i could do something) or just some type of variable like @antollika mentioned called `isNoOptions` that can be used to trigger events
Was unsure if something like that is currently available

**Comment by tonilaukka** [comment]
@peyssoa my solution was to customize NoOptionsMessage component and use useEffect to call a onSearch prop passed to parent Select component:
```
import { useEffect } from 'react'
import { components } from 'react-select'

function NoOptionsMessage(props) {
  const { inputValue, onSearch } = props.selectProps

  useEffect(() => {
    if (inputValue) {
      onSearch(inputValue)
    }
  }, [inputValue, onSearch])

  return (
    <components.NoOptionsMessage {...props}>
      {props.children}
    </components.NoOptionsMessage>
  )
}

export default NoOptionsMessage
```

We have some cases that we have more than 2000 options. If that's the case we wan't to search all the options not just the first 2000. If we have a miss in the first 2000 options then NoOptionsMessage component will call onSearch function in the parent component which will fire a backend call and make a fetch with the inputValue. We also use virtualization if there is more than 2000 options in the select component to make everything run smoothly.
